features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/Alpha8ResourceCache.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.resources
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.Bitmap
+import androidx.collection.LruCache
+
+internal interface Alpha8ResourceCache : ComponentCallbacks2 {
+    /**
+     * Generates a cache key for the bitmap. This is separated from get/put so the caller
+     * can compute the key once and reuse it for both operations.
+     * @return The cache key, or null if signature generation failed
+     */
+    fun generateKey(bitmap: Bitmap): Alpha8CacheKey?
+
+    fun get(key: Alpha8CacheKey): String?
+
+    fun put(key: Alpha8CacheKey, resourceId: String)
+}
+
+internal data class Alpha8CacheKey(
+    val width: Int,
+    val height: Int,
+    val signature: Long
+)
+
+internal class DefaultAlpha8ResourceCache(
+    private val signatureGenerator: BitmapSignatureGenerator,
+    private val cache: LruCache<Alpha8CacheKey, ByteArray> =
```

Comment:
cannot it be just `LruCache<Alpha8CacheKey, ByteArray>`? this way we would avoid making `toString` / `toByteArray` conversions.